proxy/http/HttpTransactHeaders.cc

```diff
@@ -96,6 +96,12 @@ HttpTransactHeaders::is_method_idempotent(int method)
           method == HTTP_WKSIDX_HEAD || method == HTTP_WKSIDX_PUT || method == HTTP_WKSIDX_OPTIONS || method == HTTP_WKSIDX_TRACE);
 }
 
+bool
+HttpTransactHeaders::is_method_early_data_safe(int method)
+{
+  return (method == HTTP_WKSIDX_GET || method == HTTP_WKSIDX_HEAD);
```

Comment:
@bryancall I can live with allowing only GET and HEAD. But we don't follow the RFC 8470 Section 3 and RFC 7231 Section 4.2.1 ?

RFC 8470 Section 3 says below

> The request method's safety ([RFC7231], Section 4.2.1) is one way to determine this.

https://tools.ietf.org/html/rfc8470#section-3

RFC 7231 Section 4.2.1 says below

> Of the request methods defined by this specification, the GET, HEAD,
   OPTIONS, and TRACE methods are defined to be safe.

https://tools.ietf.org/html/rfc7231#section-4.2.1

Reply:
from a customer point of view, it's better to follow the standard. We get questions every now and then about why our H/2 implementation is not fully compatible with the standard.

Reply:
Ok, lets get rid of this new method safety API, and use HttpTransactHeaders::is_method_safe(int method.

Reply:
To implement fancier control, such as discussed in https://tools.ietf.org/html/rfc8470#section-3, but we should consider that as well.

Reply:
@calavera Could you file an issue? H2 has miss behavior?
> We get questions every now and then about why our H/2 implementation is not fully compatible with the standard.